newIDE/app/src/SceneEditor/ExtractAsCustomObjectDialog.js

```diff
@@ -0,0 +1,194 @@
+        <DialogPrimaryButton
+          key="apply"
+          label={<Trans>Move instances</Trans>}
+          primary={true}
+          disabled={
+            !canCreateEventsBasedObject(extensionName, eventsBasedObjectName)
+          }
+          onClick={apply}
+        />,
+      ]}
+      secondaryActions={[
+        <HelpButton
+          helpPagePath="/objects/custom-objects-prefab-template/"
+          key="help"
+        />,
+      ]}
+      onRequestClose={onCancel}
+      onApply={apply}
+      open
+      maxWidth="sm"
+      exceptionallyStillAllowRenderingInstancesEditors
+    >
+      <ColumnStackLayout noMargin>
+        <Text>
+          <Trans>
+            Selected instances will be moved to a new custom object.
+          </Trans>
+        </Text>
+        <ResponsiveLineStackLayout noMargin expand>
+          <SelectField
+            floatingLabelText={
+              <Trans>Extension (storing the custom object)</Trans>
+            }
+            value={
+              isNewExtension ? CREATE_NEW_EXTENSION_PLACEHOLDER : extensionName
+            }
+            onChange={(e, i, extensionName) => {
+              if (extensionName === CREATE_NEW_EXTENSION_PLACEHOLDER) {
+                setNewExtension(true);
+                setExtensionName('');
+              } else {
+                setNewExtension(false);
+                setExtensionName(extensionName);
+              }
+            }}
+            fullWidth
+          >
+            {eventsFunctionsExtensions.map(eventsFunctionsExtension => (
+              <SelectOption
+                key={eventsFunctionsExtension.getName()}
+                value={eventsFunctionsExtension.getName()}
+                label={
+                  eventsFunctionsExtension.getFullName() ||
+                  eventsFunctionsExtension.getName()
+                }
+              />
+            ))}
+            <SelectOption
+              value={CREATE_NEW_EXTENSION_PLACEHOLDER}
+              label={t`<Create a New Extension>`}
+            />
+          </SelectField>
+          {isNewExtension ? (
+            <SemiControlledTextField
+              commitOnBlur
+              value={extensionName}
+              floatingLabelText={<Trans>New extension name</Trans>}
+              onChange={(extensionName: string) =>
+                setExtensionName(extensionName)
+              }
+              fullWidth
+              errorText={
+                project.hasEventsFunctionsExtensionNamed(extensionName) ? (
+                  <Trans>
+                    This name is already taken by another extension.
+                  </Trans>
+                ) : !gd.Project.isNameSafe(extensionName) ? (
+                  <Trans>
+                    This name is not valid. Only use alphanumeric characters
+                    (0-9, a-z) and underscores.
+                  </Trans>
+                ) : (
+                  undefined
+                )
+              }
+            />
```

Comment:
This approach of showing error and blocking the user is now a bit outdated and replaced in the rest of the app by just giving a fixed version.
Use `gd.Project.getSafeName` + `newNameGenerator` (search for `safeAndUniqueNewName` for examples)